src/Build/Evaluation/ItemSpec.cs

```diff
@@ -186,6 +189,15 @@ public IEnumerable<ItemFragment> FragmentsMatchingItem(string itemToMatch, out i
 
             return result;
         }
+
+        /// <summary>
+        /// Return an MSBuildGlob that represents this ItemSpec.
+        /// </summary>
+        /// <returns></returns>
+        public IMSBuildGlob ToMSBuildGlob()
+        {
+            return new CompositeGlob(Fragments.Select(f => f.ToMSBuildGlob()).ToImmutableArray());
```

Comment:
This calling pattern makes me think a ctor should take an `IEnumerable` and immutable-ize it itself.